pkg/clustermesh/clustermesh.go

```diff
@@ -110,6 +110,16 @@ func GetClusterConfig(clusterName string, backend kvstore.BackendOperations) (*c
 	return &config, nil
 }
 
+// IsClusterConfigRequired returns whether the remote kvstore guarantees that the
+// cilium cluster config will be eventually created.
+func IsClusterConfigRequired(backend kvstore.BackendOperations) (bool, error) {
```

Comment:
Why can't we propagate `DoFunc` context to here?

Reply:
Updated. While I was there, I've also added the context propagation to the Get/SetClusterConfig helpers.